### ISSUE TITLE
Style/ArrayIntersect causes unsafe fix when block is passed to any?

### ISSUE THREAD

**Issue by marvinthepa** [rubocop/rubocop#12171]
## Expected behavior

I don't really know. Either Style/ArrayIntersect needs to be marked "unsafe", or it needs to not trigger if a block is passed to `any?` ?

## Actual behavior

consider the following code:

```ruby
irb(main):003:0> ([1] & [1,2]).any? { |x| false }
=> false
```

This triggers the Style/ArrayIntersect cop, which is marked safe. When running with `--autocorrect`, this will be changed to

```ruby
irb(main):002:0> [1].intersect?([1,2]) { |x| false }
=> true
```

which does not produce the same result (see above).
Passing a block to `intersect?` is undocumented, so I guess it ignores the block and just checks the array for common elements - but passing a block to `any?` is perfectly valid.

## Steps to reproduce the problem

see above

## RuboCop version

```
$ [bundle exec] rubocop -V
1.42.0 (using Parser 3.1.3.0, rubocop-ast 1.24.1, running on ruby 3.1.3) [x86_64-linux]
```
